worlds/messenger/__init__.py

```diff
@@ -48,21 +49,28 @@ class MessengerWorld(World):
     base_offset = 0xADD_000
     item_name_to_id = {item: item_id
                        for item_id, item in enumerate(ALL_ITEMS, base_offset)}
+    mega_shard_locs = [shard for region in MEGA_SHARDS for shard in MEGA_SHARDS[region]]
     location_name_to_id = {location: location_id
-                           for location_id, location in enumerate([*ALWAYS_LOCATIONS, *SEALS], base_offset)}
+                           for location_id, location in
+                           enumerate([
+                               *ALWAYS_LOCATIONS,
+                               *SEALS,
+                               *mega_shard_locs,
+                               *BOSS_LOCATIONS,
+                           ], base_offset)}
 
-    data_version = 1
+    data_version = 2
+    required_client_version = (0, 4, 0)
```

Comment:
This is gonna be a problem. Protocol version `(0, 4, 0)` is required to support data package checksum, but your Mod does not. Please change the mod to send `(0, 3, 9)` instead until the .net lib is updated.
```suggestion
    required_client_version = (0, 3, 9)
```